firmware/examples/PWM/source/main.cpp

```diff
@@ -0,0 +1,36 @@
+#include "L0_LowLevel/delay.hpp"
+#include "L1_Drivers/gpio.hpp"
+#include "L1_Drivers/pwm.hpp"
+#include "L2_Utilities/debug_print.hpp"
+
+int main(void)
+{
+    DEBUG_PRINT("Pwm Application Starting...");
+
+    DEBUG_PRINT("Creating Gpio powered LED...");
+    DEBUG_PRINT("This Gpio powered LED represents the maximum brightness");
+    DEBUG_PRINT("and is used as reference against the Pwm powered LED");
+    DEBUG_PRINT("during its sweep.");
+    // Connect a 470 ohm resistor between the power line and the anode
+    // of the LED. The cathode is then connected to the gpio and pwm pin.
+    Gpio p4_28(4, 28);
+    p4_28.SetAsOutput();
+    p4_28.SetLow();
+
+    DEBUG_PRINT("Creating Pwm powered LED...");
+    Pwm p2_0 = Pwm::CreatePwm<1>();
+    // Initialize Pwm at 1,000 hz
+    p2_0.Initialize(1'000);
+    float duty = 0;
+
+    while (1)
+    {
+        for (int i = 0; i <= 10; i++)
```

Comment:
Can you make this more granular? Like between 0 -> 255, and decrease the Delay down to maybe 10ms or lower? This will make the fading effect look more smooth.